lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -177,8 +177,9 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitBinaryExpression(BinaryExpression node) {
-    final isDartCoreBoolean = node.staticType.isDartCoreBool;
-    if (!isDartCoreBoolean ||
+    final isDartCoreBoolean = node.staticType?.isDartCoreBool;
+    if (isDartCoreBoolean == null ||
+        !isDartCoreBoolean ||
```

Comment:
For NNBD, I suspect we'll want to change this to
```
final isDartCoreBoolean = node.staticType?.isDartCoreBool ?? false;
  if (!isDartCoreBoolean ||
```
Otherwise, I think line 182 will have to read `!isDartCoreBoolean!`, which, while nicely emphatic, is a bit much.

Reply:
Haha.  Thanks!